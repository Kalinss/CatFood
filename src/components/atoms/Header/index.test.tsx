import React from "react";
import { Header } from "./index";
import renderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Header", () => {
  it("should be rendered", () => {
    const tree = renderer.create(<Header as="h1">123</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("check prop as by default", () => {
    const wrapper = shallow(<Header>123</Header>);
    expect(wrapper.exists("span")).toBe(true);
  });

  it("should be render children string", () => {
    const wrapper = shallow(<Header>123</Header>);
    expect(wrapper.find("span").text()).toStrictEqual("123");
  });

  it("check prop as", () => {
    const wrapper = shallow(<Header as="h1">123</Header>);
    expect(wrapper.exists("h1")).toBe(true);
  });

  it("render correct prop 'as' h1", () => {
    const wrapper = shallow(<Header as="h1">123</Header>);
    expect(wrapper.exists("h1")).toBe(true);
  });

  it("render correct prop 'as' h2", () => {
    const wrapper = shallow(<Header as="h2">123</Header>);
    expect(wrapper.exists("h2")).toBe(true);
  });

  it("render correct prop 'as' h3", () => {
    const wrapper = shallow(<Header as="h3">123</Header>);
    expect(wrapper.exists("h3")).toBe(true);
  });

  it("render correct prop 'as' h4", () => {
    const wrapper = shallow(<Header as="h4">123</Header>);
    expect(wrapper.exists("h4")).toBe(true);
  });

  it("render correct prop 'as' h5", () => {
    const wrapper = shallow(<Header as="h5">123</Header>);
    expect(wrapper.exists("h5")).toBe(true);
  });

  it("render correct prop 'as' h6", () => {
    const wrapper = shallow(<Header as="h6">123</Header>);
    expect(wrapper.exists("h6")).toBe(true);
  });

  it("component should be have class", () => {
    const wrapper = shallow(
      <Header as="h6" classStyle="asd">
        123
      </Header>
    );
    expect(wrapper.hasClass("asd")).toBe(true);
  });
});
