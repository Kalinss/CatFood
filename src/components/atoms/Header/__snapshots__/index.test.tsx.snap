// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be rendered 1`] = `
<div
  className="header"
>
   
  <h1>
    123
  </h1>
   
</div>
`;
